Document v3 custom partitions (#5309)
influxdata/docs-v2: content/influxdb/cloud-dedicated/reference/internals/storage-engine.md

- Processes line protocol and persists time series data to the
  [Object store](#object-store) in Apache Parquet format. Each Parquet file
  represents a _partition_--a logical grouping of data.
- Makes [yet-to-be-persisted](/influxdb/cloud-dedicated/reference/internals/durability/#data-ingest) data available to [Queriers](#querier) to ensure
  leading edge data is included in query results.
- Maintains a short-term [write-ahead log (WAL)](/influxdb/cloud-dedicated/reference/internals/durability/) to prevent data loss in case of a
  service interruption.

##### Ingester scaling strategies

The Ingester can be scaled both [vertically](#vertical-scaling) and
[horizontally](#horizontal-scaling).
Horizontal scaling increases write throughput and is typically the most
effective scaling strategy for the Ingester.

### Querier

The Querier handles query requests and returns query results for requests.
It supports both SQL and InfluxQL through
[Apache Arrow DataFusion](https://arrow.apache.org/datafusion/user-guide/introduction.html).

#### Query life cycle

At query time, the querier:

1.  Receives the query request and builds a query plan.

> 2.  Queries the [Ingester](#ingester) to:

Reviewer: I wonder if you want to make `Ingester` her plural because the Querier (the one that handle the query) talks to all ingesters in the cluster for not-yet-persisted data

The Ingester can be scaled both [vertically](#vertical-scaling) and
[horizontally](#horizontal-scaling).
Horizontal scaling increases write throughput and is typically the most
effective scaling strategy for the Ingester.

### Querier

The querier handles query requests and returns query results to the client that
mad the query request.
It supports both SQL and InfluxQL through
[Apache Arrow DataFusion](https://arrow.apache.org/datafusion/user-guide/introduction.html).

#### Query life cycle

At query time, the querier:

1.  Receives the query request and builds a query plan.
2.  Queries the [Ingester](#ingester) to:

    - ensure the schema assumed by the query plan matches the schema of written data.
    - include recently written, yet-to-be-persisted data in query results.

3.  Queries the [Catalog](#catalog) to identify the physical location of partitions
    that contain the queried data.
4.  Reads partition Parquet files from the [Object store](#object-store)
    containing the queried data and scans each row to find those that match the
    logic defined in the query plan.
5.  Returns the data specified in the query and performs any additional

>     operations specified in the query plan.

Reviewer 2: ```suggestion
    operations (for example: deduplicating, merging, and sorting) specified in the query plan.
```

6.  Returns the query result to the client.

##### Querier scaling strategies

The Querier can be scaled both [vertically](#vertical-scaling) and
[horizontally](#horizontal-scaling).
Horizontal scaling increases query throughput to handle more concurrent queries.

> Vertical scaling improves the queriers ability to process computation-heavy queries.

Reviewer 2: ```suggestion
Vertical scaling improves the Querier's ability to process computationally intensive queries.
```

### Catalog

The Catalog is a PostgreSQL-compatible relational database that stores metadata
related to your time series data including schema information and physical
locations of partitions in the [Object store](#object-store).
It fulfills the following roles:

- Provides information about the schema of written data.
- Tells the [Ingester](#ingester) what partitions to persist data to.
- Tells the [Querier](#querier) what partitions contain the queried data.

##### Catalog scaling strategies

Scaling strategies available for the Catalog depend on the PostgreSQL-compatible
database used to run the catalog. All support [vertical scaling](#vertical-scaling).
Most support [horizontal scaling](#horizontal-scaling) for redundancy and failover. 

### Object store

The Object store contains time series data in [Apache Parquet](https://parquet.apache.org/) format.

> Each parquet file represents a partition.

Reviewer 2: ```suggestion
Each Parquet file represents a partition.
```
